apps/condo/domains/user/oidc/adapter.js

```diff
@@ -0,0 +1,139 @@
+const Redis = require('ioredis')
+const { isEmpty } = require('lodash')
+
+const GRANTABLE = new Set([
+    'AccessToken',
+    'AuthorizationCode',
+    'RefreshToken',
+    'DeviceCode',
+    'BackchannelAuthenticationRequest',
+])
+
+const CONSUMABLE = new Set([
+    'AuthorizationCode',
+    'RefreshToken',
+    'DeviceCode',
+    'BackchannelAuthenticationRequest',
+])
+
+// OIDC Redis connection! initialized if required to avoid build time redis connections!
+let REDIS = null
+
+function grantKeyFor (id) {
+    return `grant:${id}`
+}
+
+function userCodeKeyFor (userCode) {
+    return `userCode:${userCode}`
+}
+
+function uidKeyFor (uid) {
+    return `uid:${uid}`
+}
+
+class AdapterFactory {
+    constructor (name) {
+        // TODO(pahaz): store Client inside keystone gql and others models inside Redis!
+        return new RedisAdapter(name)
+    }
+}
+
+class RedisAdapter {
```

Comment:
Maybe to extract it to `RedisAdapter.js` and let this `adapter.js` module be `adapter/index.js`?